R/guides-.R

```diff
@@ -529,11 +528,28 @@ Guides <- ggproto(
     grobs <- vector("list", length(guides))
     for (i in seq_along(grobs)) {
       grobs[[i]] <- guides[[i]]$draw(
-        theme = theme, position = as.character(positions[i]),
+        theme = theme, position = positions[i],
         direction = directions[i], params = params[[i]]
       )
+      # we'll merge inside legends with same coordinate into same guide box
+      # here, we define the groups of the inside legends
+      if (identical(positions[i], "inside")) {
+        positions[i] <- paste(
+          "inside",
+          paste(attr(.subset2(grobs, i), "inside_position"), collapse = "_"),
+          sep = "_"
+        )
+      }
     }
-    keep <- !vapply(grobs, is.zero, logical(1))
+
+    # move inside legends to the last
+    positions <- factor(positions, 
+      levels = c(.trbl, unique(positions[startsWith(positions, "inside")]))
+    )
+    keep <- !vapply(grobs, is.zero, logical(1), USE.NAMES = FALSE)
+
+    # we grouped the legends by the positions
+    # for inside legends, they'll be splitted by the actual inside coordinate
     split(grobs[keep], positions[keep])
```

Comment:
I think for this PR it might be easier to lift the reponsibilities of managing the guide position from the `Guides$draw()` method to the `Guides$assemble()` method. The assemble method can then properly forward the positions and 'position_inside' information to the `Guides$package_box()` method.